src/cascadia/QueryExtension/ILLMProvider.idl

```diff
@@ -0,0 +1,25 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+namespace Microsoft.Terminal.Query.Extension
+{
+    interface ILLMProvider
+    {
+        void ClearMessageHistory();
+        void SetSystemPrompt(String systemPrompt);
+        void SetContext(IContext context);
+
+        Windows.Foundation.IAsyncOperation<IResponse> GetResponseAsync(String userPrompt);
+    }
+
+    interface IResponse
+    {
+        String Message { get; };
+        Boolean IsError { get; };
```

Comment:
Are there scenarios where an extension would want to report more verbose error details (i.e., beyond communicating that a failure has occurred)?

Reply:
There potentially could be, which for now I would expect they could just put in `Message`. If in the future we need more (such as an error code or something) we could add it to this interface